worker-document/src/main/java/com/hpe/caf/worker/document/impl/ScriptImpl.java

```diff
@@ -242,6 +246,17 @@ public void uninstall()
     @Override
     public void unload()
     {
+        unloadScriptBindings();
+    }
+
+    private void unloadScriptBindings() {
+        if (loadedScriptBindings instanceof AutoCloseable) {
+            try {
+                ((AutoCloseable) loadedScriptBindings).close();
+            } catch (final Exception ex) {
```

Comment:
What types of exceptions are possible here?
Do we need special handling for runtime or `InterruptedException`s for example?
```suggestion
            } catch (final RuntimeException ex) {
                throw ex;
            } catch (final InterruptedException ex) {
                LOG.warn("The thread was interruptted whilst closing the script bindings", ex);
                Thread.currentThread().interrupt();
            } catch (final Exception ex) {
```

Reply:
Looking at the GraalJSBindings close method:

https://github.com/graalvm/graaljs/blob/master/graal-js/src/com.oracle.truffle.js.scriptengine/src/com/oracle/truffle/js/scriptengine/GraalJSBindings.java#L175

it calls close on a context object:

https://www.graalvm.org/sdk/javadoc/org/graalvm/polyglot/Context.html#close--

which can throw these 2 exceptions:

PolyglotException - in case the close failed due to a guest language error.
IllegalStateException - if the context is currently executing on another thread

I can't find where RuntimeException and InterruptedException are thrown from though?

Reply:
Oh, just checked, PolyglotException and IllegalStateException are RuntimeExceptions